src/Extensions/InputSystem/InputGameEngineComponent.ts

```diff
@@ -0,0 +1,27 @@
+import { GameEngineComponent } from "../../Core/GameEngineComponent.ts";
+import { Device } from "./Device.ts";
+
+/**
+ * A game engine component that manages input devices.
+ * @extends GameEngineComponent
+ * Member of unofficial namespace InputSystem.
+ */
+export class InputGameEngineComponent extends GameEngineComponent {
+  private _devices: Device[] = [];
+
+  /**
+   * Gets a device of a specified class inheriting Device.
+   */
+  public getDevice<T extends Device>(deviceClass: new () => T): T | null {
```

Comment:
Probably should be abstract ctor instead (i will probably need to explain further that one)